datadog_lambda/wrapper.py

```diff
@@ -103,33 +107,42 @@ def __init__(self, func):
     def __call__(self, event, context, **kwargs):
         """Executes when the wrapped function gets called"""
         self._before(event, context)
+        response = None
         try:
-            return self.func(event, context, **kwargs)
+            response = self.func(event, context, **kwargs)
+            return response
         except Exception:
             submit_errors_metric(context)
             if self.span:
                 self.span.set_traceback()
             raise
         finally:
-            self._after(event, context)
+            self._after(event, context, response)
 
     def _before(self, event, context):
         try:
 
             set_cold_start()
             submit_invocations_metric(context)
-            # Extract Datadog trace context from incoming requests
-            dd_context = extract_dd_trace_context(event)
+            # Extract trigger tags and add them to a subsegment if found
```

Comment:
nits: I would shift some of the logic to an intermediary function, say `create_dd_subsegments`, to keep this entry function better manifest the high-level logic and not overloaded/overwhelmed by details. 

```python
trigger_tags = extract_trigger_tags(event, context)
dd_context, trace_context_source = extract_dd_trace_context(event)
create_dd_subsegments(trigger_tags, dd_context, trace_context_source)
```

BTW, perhaps by doing so, you could even avoid creating **two** subsegments (is it necessary?)? ie., `create_dd_subsegment` creates one subsegment, and add the trace context or root span tags based on the need?